bindings/rust/s2n-tls/src/testing/s2n_tls.rs

```diff
@@ -649,4 +649,53 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn system_certs_loaded_by_default() {
+        let keypair = CertKeyPair::default();
+
+        // Load the server certificate into the trust store by overriding the OpenSSL default
+        // certificate location.
+        temp_env::with_var("SSL_CERT_FILE", Some(keypair.cert_path()), || {
+            let mut builder = Builder::new();
+            builder
+                .load_pem(keypair.cert(), keypair.key())
+                .unwrap()
+                .set_security_policy(&security::DEFAULT_TLS13)
+                .unwrap()
+                .set_verify_host_callback(InsecureAcceptAllCertificatesHandler {})
+                .unwrap();
+
+            let config = builder.build().unwrap();
+            establish_connection(config);
+        });
+    }
+
+    #[test]
+    fn disable_loading_system_certs() {
+        let keypair = CertKeyPair::default();
+
+        // Load the server certificate into the trust store by overriding the OpenSSL default
+        // certificate location.
+        temp_env::with_var("SSL_CERT_FILE", Some(keypair.cert_path()), || {
+            let mut builder = Builder::new();
+            builder
+                .load_pem(keypair.cert(), keypair.key())
+                .unwrap()
+                .set_security_policy(&security::DEFAULT_TLS13)
+                .unwrap()
+                .set_verify_host_callback(InsecureAcceptAllCertificatesHandler {})
+                .unwrap();
+
+            // Disable loading system certificates
+            builder.with_system_certs(false).unwrap();
+
+            let config = builder.build().unwrap();
+            let mut pair = tls_pair(config);
+
+            // System certificates should not be loaded into the trust store. The handshake
+            // should fail since the cert should not be trusted.
+            assert!(poll_tls_pair_result(&mut pair).is_err());
```

Comment:
Can we confirm the reason for the failure? Either check the error name (pretty fragile :/) or retry the handshake after adding the cert to the trust store?